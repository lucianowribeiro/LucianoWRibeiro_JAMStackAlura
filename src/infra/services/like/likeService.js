import isStagingEnv from '../../env/isStagingEnv';
import HttpClient from '../../http/HttpClient';
import authService from '../auth/authService';

const BASE_URL = isStagingEnv
  ? 'https://instalura-api-git-master-omariosouto.vercel.app'
  : 'https://instalura-api-omariosouto.vercel.app';

const likeService = {
  async like(id, data, HttpClientModule = HttpClient) {
    try {
      const token = await authService().getToken();
      const response = await HttpClientModule(`${BASE_URL}/api/posts/${id}/like`, {
        method: 'POST',
        body: { data },
        headers: {
          Authorization: `Bearer ${token}`,
        },
      });
      return response;
    } catch (error) {
      return null;
    }
  },

};

export default likeService;
