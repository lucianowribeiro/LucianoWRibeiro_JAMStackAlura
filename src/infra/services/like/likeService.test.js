import likeService from './likeService';

const dataFake = {
  _id: 'some id',
  photoUrl: 'https://images.endeavor.org.br/uploads/2015/05/simples-nacional1.jpg',
  description: 'some description',
  filter: 'some filter',
  user: 'some user',
  likes: [
    {
      _id: 'some id',
      user: 'some user',
    },
  ],
  createdAt: 'some data',
  updatedAt: 'some data 2',
  __v: 0,
};

async function HttpClientModule() {
  return {
    data: dataFake,
  };
}

describe('likeService', () => {
  describe('like()', () => {
    describe('when click on image', () => {
      describe('and succeed', () => {
        test('setting the number of likes', async () => {
          const { data } = await likeService.like('some id', 1, HttpClientModule);
          await expect(data).toMatchSnapshot();
          await expect(data.likes.length).toBe(dataFake.likes.length);
        });
      });
      describe('and it fails', () => {
        test('return null and keeping the same number of likes', async () => {
          const response = await likeService.like('some id', {});
          await expect(response).toMatchSnapshot();
          await expect(response).toBeNull();
        });
      });
    });
  });
});
