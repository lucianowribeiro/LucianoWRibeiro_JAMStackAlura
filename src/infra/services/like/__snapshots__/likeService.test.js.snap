// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`likeService like() when click on image and it fails return null and keeping the same number of likes 1`] = `null`;

exports[`likeService like() when click on image and succeed setting the number of likes 1`] = `
Object {
  "__v": 0,
  "_id": "some id",
  "createdAt": "some data",
  "description": "some description",
  "filter": "some filter",
  "likes": Array [
    Object {
      "_id": "some id",
      "user": "some user",
    },
  ],
  "photoUrl": "https://images.endeavor.org.br/uploads/2015/05/simples-nacional1.jpg",
  "updatedAt": "some data 2",
  "user": "some user",
}
`;
